test/weather_man.spec.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import fetchMock from 'fetch-mock';
+import { WeatherMan } from '../src/weather_man';
+import { flushPromises } from '../src/api';
+
+let weatherMan
+
+beforeEach(() => {
+  fetch.mockResponse(JSON.stringify({name: "Padua", weather: [{description: "sunny"}]}), {status: 200});
+  weatherMan = mount(<WeatherMan />);
+});
+
+test('renders without crashing', () => {
+  mount(<WeatherMan />);
+})
+
+test('renders a heading with hardcoded state values for city and weather without API call', () => {
+  weatherMan.setState({city: "Padua", weather: "sunny"})
```

Comment:
Just a small point - I would probably not test setState on the component directly here because it's not really part of the component's public interface (at least at the moment in this example).

Instead maybe you could have two tests with different mockResponse data that would show that whatever is in the mockResponse ends up in the h1?